beacon-chain/operations/attestations/kv/seen_bits.go

```diff
@@ -19,7 +19,16 @@ func (p *AttCaches) insertSeenBit(att *ethpb.Attestation) error {
 		if !ok {
 			return errors.New("could not convert to bitlist type")
 		}
-		seenBits = append(seenBits, att.AggregationBits)
+		alreadyExists := false
+		for _, bit := range seenBits {
+			if bit.Len() == att.AggregationBits.Len() && bit.Contains(att.AggregationBits) {
+				alreadyExists = true
+				break
```

Comment:
if it already exists we should exit early then

Reply:
I deliberately do not exit -- and update the expiry time, so the cache remembers given attestation longer (supposing that if we had one duplicate, there may be another coming). Thoughts?

Reply:
Dont feel too strongly on it if the purpose is to keep it alive in the cache longer. I am not too familiar with the code here so maybe @terencechain can give a better opinion on this.

Reply:
Agree! @terencechain , please, let me know if that updating of expiry time is even necessary? I did it this way assuming when we had one duplicate we might get more and expiry is to be tracked from the time of the last duplicate received.

If that update to expiry time is unnecessary, then Nishant's suggestion to exit early is spot on.

Reply:
Both are likely fine. Updating the expiry time is more lenient in terms of validator profitability in some extreme cases and the performance impact is negligible. Hence I'm leaning towards `update` versus `exit`.